yaks/lib/yaks/reader/hal.rb

```diff
@@ -29,7 +29,7 @@ def convert_links(links)
             rel = rel.to_sym if rel =~ /\A\w+\z/
             Resource::Link.new(rel: rel, uri: l['href'], options: options)
           end
-        end.to_set
+        end
```

Comment:
This has nothing to do with HAL. The `links` on a `Yaks::Resource` could be considered conceptually a set, because the ordering is undefined (i.e. they might have a certain order but it's not something you should rely on). I guess when I wrote this code I had that on my mind. 

In early versions of Yaks I actually used [Hamster](https://github.com/hamstergem/hamster) data structures for these things, since most things in Yaks are immutable, but in the end decided to keep it simple and just use Ruby arrays and hashes.